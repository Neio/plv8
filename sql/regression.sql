-- https://github.com/plv8/plv8/issues/527
DO $$
  plv8.prepare('SELECT * FROM (VALUES (1)) AS foo').cursor().fetch.apply(null)
$$ language plv8;

-- https://github.com/plv8/plv8/issues/540
CREATE FUNCTION test_array_buffer_int8_offset() RETURNS bytea AS $$
	let ab = new ArrayBuffer(3);
	let view = new Uint8Array(ab);
	view[0] = 6;
	view[1] = 9;
  view[2] = 12;
	
	let r = new Uint8Array(ab, 1);
	return r; 
$$ LANGUAGE plv8;

SELECT test_array_buffer_int8_offset();

CREATE FUNCTION test_array_buffer_int16_offset() RETURNS bytea AS $$
	let ab = new ArrayBuffer(6);
	let view = new Uint16Array(ab);
	view[0] = 6;
	view[1] = 9;
  view[2] = 12;
	
	let r = new Uint16Array(ab, 2);
	return r; 
$$ LANGUAGE plv8;

SELECT test_array_buffer_int16_offset();

CREATE FUNCTION test_array_buffer_int32_offset() RETURNS bytea AS $$
	let ab = new ArrayBuffer(12);
	let view = new Uint32Array(ab);
	view[0] = 6;
	view[1] = 9;
  view[2] = 12;
	
	let r = new Uint32Array(ab, 4);
	return r; 
$$ LANGUAGE plv8;

SELECT test_array_buffer_int32_offset();
