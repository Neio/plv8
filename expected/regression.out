-- https://github.com/plv8/plv8/issues/527
DO $$
  plv8.prepare('SELECT * FROM (VALUES (1)) AS foo').cursor().fetch.apply(null)
$$ language plv8;
ERROR:  cannot find cursor
CONTEXT:  undefined() LINE 2:   plv8.prepare('SELECT * FROM (VALUES (1)) AS foo').cursor().fetch.apply(null)
-- https://github.com/plv8/plv8/issues/540
CREATE FUNCTION test_array_buffer_int8_offset() RETURNS bytea AS $$
	let ab = new ArrayBuffer(3);
	let view = new Uint8Array(ab);
	view[0] = 6;
	view[1] = 9;
  view[2] = 12;
	
	let r = new Uint8Array(ab, 1);
	return r; 
$$ LANGUAGE plv8;
SELECT test_array_buffer_int8_offset();
 test_array_buffer_int8_offset 
-------------------------------
 \x090c
(1 row)

CREATE FUNCTION test_array_buffer_int16_offset() RETURNS bytea AS $$
	let ab = new ArrayBuffer(6);
	let view = new Uint16Array(ab);
	view[0] = 6;
	view[1] = 9;
  view[2] = 12;
	
	let r = new Uint16Array(ab, 2);
	return r; 
$$ LANGUAGE plv8;
SELECT test_array_buffer_int16_offset();
 test_array_buffer_int16_offset 
--------------------------------
 \x09000c00
(1 row)

CREATE FUNCTION test_array_buffer_int32_offset() RETURNS bytea AS $$
	let ab = new ArrayBuffer(12);
	let view = new Uint32Array(ab);
	view[0] = 6;
	view[1] = 9;
  view[2] = 12;
	
	let r = new Uint32Array(ab, 4);
	return r; 
$$ LANGUAGE plv8;
SELECT test_array_buffer_int32_offset();
 test_array_buffer_int32_offset 
--------------------------------
 \x090000000c000000
(1 row)

